src/transaction/log_page_buffer.c

```diff
@@ -11403,11 +11405,34 @@ logpb_remove_all_in_log_path (THREAD_ENTRY * thread_p, const char *db_fullname,
 
   fileio_unformat (thread_p, log_Name_active);
   fileio_unformat (thread_p, log_Name_info);
-  fileio_unformat (thread_p, log_Name_metainfo);
+  logpb_delete_metainfo_files_internal (thread_p, log_Path, log_Prefix);
 
   return NO_ERROR;
 }
 
+/*
+ * logpb_delete_metainfo_files_internal - delete all existing metainfo volumes found in the log path
+ *
+ * NOTE: a transaction server with remote storage has its own specific metalog file;
+ *      in 'single-node' install mode - where both transaction server and page server
+ *      execute on the same machine, both the regular metalog volume and the transaction server specific
+ *      metalog volume are located in the same directory
+ */
+void
+logpb_delete_metainfo_files_internal (THREAD_ENTRY * thread_p, const char *logpath, const char *prefix_logname)
+{
+  char metainfo_file_name[PATH_MAX];
+  const bool is_ts_with_remote_storage = is_tran_server_with_remote_storage ();
```

Comment:
Regardless of the value of the is_ts_with_remote_storage variable, this function always seems to call fileio_make_log_metainfo_name(true) and fileio_make_log_metainfo_name(false) once. Is there any reason to use this variable?

Reply:
You are, somewhat, right.

The only difference is in relation with the order in which deletes happen.
If the database files are deleted on a transaction server:
- first delete FILEIO_SUFFIX_LOGMETA_TRAN   "_lgmeta_tran"
- then delete FILEIO_SUFFIX_LOGMETA        "_lgmeta"
If on a page server, the other way around.
The comments also specifies that the "own" metalog file is first deleted, then the other.

If there was a way to mandate to the `fileio_unformat` - `fileio_unformat_and_rename` that the files are mandatory to exist or not, it would make even more sense.

As a side comment, there are two log meta files to be able to execute the "single_node" configuration where both servers have the files in the same location and use the same config file(s).